File: 0271-Encode and Decode Strings/0271-Encode and Decode Strings.cpp
class Codec {
public:

    // Encodes a list of strings to a single string.
    string encode(vector<string>& strs) {
        string result;
        for (string& str : strs) {
            result += to_string(str.size()) + '@' + str;
        }
        return result;
    }

    // Decodes a single string to a list of strings.
    vector<string> decode(string s) {
        vector<string> result;
        int i = 0;
        while (i < s.size()) {
            int len = 0;
            while (i < s.size() && s[i] != '@') {
                len = len * 10 + (s[i] - '0');
                ++i;
            }
            
            ++i;
            result.push_back(s.substr(i, len));
            i += len;
        }

        return result;
    }
};

// Your Codec object will be instantiated and called as such:
// Codec codec;
// codec.decode(codec.encode(strs));

File: 0201-0300/0271-Encode and Decode Strings/0271-Encode and Decode Strings.py
class Codec:
    def encode(self, strs: [str]) -> str:
        """Encodes a list of strings to a single string.
        """
        return ''.join(str(len(s)) + '@' + s for s in strs)
        

    def decode(self, s: str) -> [str]:
        """Decodes a single string to a list of strings.
        """
        result = []
        i = 0
        while i < len(s):
            j = s.find('@', i)
            i = j + 1 + int(s[i: j])
            result.append(s[j + 1: i])
        
        return result


# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))
